lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -1030,7 +1030,7 @@ def load_credential_from_file(file_path, options = {})
         next
       end
 
-      unless !sname_hostname || sname_hostname.to_s.casecmp?(credential.server.components[1])
+      unless !sname_hostname || sname_hostname.to_s.downcase.ends_with?(credential.server.components[1])
```

Comment:
I think you want something like this to prevent tickets for domain `foobar.local` from being used for `bar.local`. It should either equal or you should add the separator character to the suffix you're checking.

```suggestion
      unless !sname_hostname || sname_hostname.to_s.downcase == credential.server.components[1] || sname_hostname.to_s.downcase.ends_with?('.' + credential.server.components[1])
```

Reply:
oh yep I see what you're saying yea that seems like a good shout